### ISSUE TITLE
[EN] Reducing page/block size by compressing ViewState

### ISSUE THREAD

**Issue by dcs619** [SparkDevNetwork/Rock#1441]
This may or may not turn into a PR...bear with me.  I've been watching our data performance since launch and noticed a 200-300ms increase in page load times as we've added new blocks.  We also have the opportunity to reach mobile users who may not live in a 4G area.

So would there any benefit to implementing ViewState compression?  Examples [here](http://www.codeproject.com/Articles/14733/ViewState-Compression), [here](http://stackoverflow.com/questions/1547532/how-to-minimize-viewstate-size-of-a-page-in-asp-net), and a full write-up of how ViewState works [here](http://weblogs.asp.net/infinitiesloop/Truly-Understanding-Viewstate).  The only drawback to compression seems to be server CPU cycles, which we (currently) barely touch.

It's entirely possible this approach is obsolete, as these articles are almost 10 years old.  @mikepetersonccv any thoughts?

**Comment by edmistj** [comment]
We actually been thinking about using ViewState caching. For this the ViewState is stored on the server. Still much to be research on this but it could have a good impact on page loads.

**Comment by mikepetersonccv** [comment]
When I looked into it last week, it sounds like the default https://msdn.microsoft.com/en-us/library/system.web.ui.hiddenfieldpagestatepersister(v=vs.110).aspx, which Rock currently uses, is already compressed, so that seemed like a dead end.  The alternative, https://msdn.microsoft.com/en-us/library/system.web.ui.sessionpagestatepersister(v=vs.110).aspx, reduced the hiddenfield ViewState size by 90% or more when I tried it on Rock.  

A couple of concerns with SessionPageStatePersister
1) What about load balancing?
2) What is the impact to Server Memory Usage.  For example, if 2000 sessions each had 250KB viewstates, that could 500MB of RAM.  
3) What about multiple tabs on a browser for the same session and page?

A few options we have
1) The option of HiddenField or Session ViewState page persister, is a "Per Page" option.  So, we could have it be an option on each RockPage and/or Site.
2) We could keep using the HiddenField method, but be more deliberate about how/if we use ViewState.  I'm sure there is some low-hanging fruit where we are using ViewState, but don't really need to be  (for example, Content Pages with no form entry)
